src/components/molecules/UserProfilePicture/UserProfilePicture.tsx

```diff
@@ -16,154 +14,108 @@ import {
 
 import { User } from "types/User";
 
-// @debt remove styled-components in favour of using our standard scss patterns
-import * as S from "./UserProfilePicture.styles";
+import { UserReactions } from "./UserReactions";
+
 import "./UserProfilePicture.scss";
 
+const randomAvatarUrl = (id: string) =>
+  "/avatars/" +
+  RANDOM_AVATARS[Math.floor(id?.charCodeAt(0) % RANDOM_AVATARS.length)];
+
+export interface AvatarUrlProps {
+  user: WithId<User>;
+  miniAvatars?: boolean;
+}
+
+const avatarUrl = ({ user, miniAvatars }: AvatarUrlProps): string => {
+  const { id, anonMode, pictureUrl } = user;
+
+  if (anonMode || !id) {
+    return DEFAULT_PROFILE_IMAGE;
+  }
+
+  if (!miniAvatars && pictureUrl) {
+    return pictureUrl;
+  }
+
+  // @debt how is this intended to be used? Why doesn't it use the profile image? It seems to be something set on venue config..
+  if (miniAvatars) {
+    return randomAvatarUrl(id);
+  }
+
+  return DEFAULT_PROFILE_IMAGE;
+};
+
 export interface UserProfilePictureProp {
   user: WithId<User>;
   isAudioEffectDisabled?: boolean;
   miniAvatars?: boolean;
   avatarClassName?: string;
   avatarStyle?: object;
   containerStyle?: object;
-  reactionPosition?: "right" | "left" | undefined;
+  reactionPosition?: "left" | "right";
 }
 
-// @debt This component should be divided into a few with simpler logic. Also, remove `styled components`
-// @debt the UserAvatar component serves a very similar purpose to this, we should unify them as much as possible
-const UserProfilePicture: React.FC<UserProfilePictureProp> = ({
+export const UserProfilePicture: React.FC<UserProfilePictureProp> = ({
   isAudioEffectDisabled = true,
   miniAvatars = false,
-  avatarClassName = "profile-icon",
-  avatarStyle,
-  containerStyle,
-  reactionPosition,
+  avatarClassName = "UserProfilePicture__avatar",
```

Comment:
I guess this should not have a default since `UserProfilePicture__avatar` is already used in avatarClasses

Reply:
Agreed. Not sure why I did it this way originally, might have been intending to allow users to override it completely, then changed my mind and figured that wasn't a good idea.

Reply:
fixed in https://github.com/sparkletown/sparkle/pull/1323/commits/134a48d8abe7e0a93cf6e9045b5c6bc2d57f10a5